presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriterFactory.java

```diff
@@ -303,6 +294,35 @@ else if (com.facebook.hive.orc.OrcOutputFormat.class.getName().equals(storageFor
         }
     }
 
+    @VisibleForTesting
+    OrcWriterOptions buildOrcWriterOptions(ConnectorSession session, Properties schema)
+    {
+        boolean mapStatisticsEnabled = isMapStatisticsEnabled(schema);
+        int flatMapKeyLimit = getFlatMapKeyLimit(schema);
+        Set<Integer> flattenedColumns = getFlattenedColumns(schema, session);
+
+        return orcFileWriterConfig
+                .toOrcWriterOptionsBuilder()
+                .withFlushPolicy(DefaultOrcWriterFlushPolicy.builder()
+                        .withStripeMinSize(getOrcOptimizedWriterMinStripeSize(session))
+                        .withStripeMaxSize(getOrcOptimizedWriterMaxStripeSize(session))
+                        .withStripeMaxRowCount(getOrcOptimizedWriterMaxStripeRows(session))
+                        .build())
+                .withDictionaryMaxMemory(getOrcOptimizedWriterMaxDictionaryMemory(session))
+                .withIntegerDictionaryEncodingEnabled(isIntegerDictionaryEncodingEnabled(session))
+                .withStringDictionaryEncodingEnabled(isStringDictionaryEncodingEnabled(session))
+                .withStringDictionarySortingEnabled(isStringDictionarySortingEnabled(session))
+                .withMaxStringStatisticsLimit(getOrcStringStatisticsLimit(session))
+                .withIgnoreDictionaryRowGroupSizes(isExecutionBasedMemoryAccountingEnabled(session))
+                .withDwrfStripeCacheEnabled(isDwrfWriterStripeCacheEnabled(session))
+                .withDwrfStripeCacheMaxSize(getDwrfWriterStripeCacheMaxSize(session))
+                .withFlattenedColumns(flattenedColumns)
+                .withMaxFlattenedMapKeyCount(flatMapKeyLimit)
+                .withMapStatisticsEnabled(mapStatisticsEnabled)
```

Comment:
will also include the keyword flat in this method name. I would prefer a similar naming convention for all the methods that deals with flat map properties.